### PR TITLE
fix: `RequestInit` doesn't allow explicit `undefined`

### PR DESCRIPTION
**Pull request by chlonel**
Fixes: https://github.com/nodejs/undici/issues/1225

### CODE REVIEW

**Review by mcollina — APPROVED** [review]
lgtm

File: lib/fetch/request.js [hunk 1]
```diff
@@ -127,12 +127,12 @@ class Request {
     }
 
     // 10. If init["window"] exists and is non-null, then throw a TypeError.
-    if ('window' in init && window != null) {
+    if (init.window !== undefined && window != null) {
```
Comment (targos) [inline lib/fetch/request.js:130]: shouldn't this have `init.window != null` ?

Comment (ronag) [reply lib/fetch/request.js:130]: @chlonel can you fix?

Comment (ronag) [reply lib/fetch/request.js:130]: Actually it doesn't matter here. It's the same.

Comment (targos) [reply lib/fetch/request.js:130]: It's not. `window` is a local variable created a few lines above (`let window = 'client'`).

Comment (targos) [reply lib/fetch/request.js:130]: > If init["window"] exists and is non-null, then throw a TypeError.

To me, this means that we should check if `init["window"]` in non-null, not `window`. But maybe the spec or the comment is wrong.

Comment (ronag) [reply lib/fetch/request.js:130]: Ah, sorry I missed that

Comment (ronag) [reply lib/fetch/request.js:130]: [Fixed](https://github.com/nodejs/undici/commit/ec2813498dac2534a53590b5e81d92c3ce334597)

**Pull request merged**